collection_script_draft1.py

```diff
@@ -0,0 +1,409 @@
+        attempt_count += 1
+    # sleeps for 0.5 seconds for the above operation to complete
+    time.sleep(0.5)
+
+    # keep looping unless the id of the current id is the same as the argument passed to the function
+    # for authentication purposes only.
+    while str(ad_id) == str(movie_id):
+        ad_id = driver.execute_script(
+            'return document.getElementsByClassName("html5-video-info-panel-content")[0].children[0].children[1].textContent.replace(" ","").split("/")[0]'
+        )
+
+    # returns the id of the add, logical value for if the ad is skippable, the duration of the ad to skip the video (up to 6 seconds long), and the start
+    # resolution of the ad
+    return ad_id, skippable_add, skip_duration, start_resolution
+
+
+def driver_code(driver: webdriver.Chrome):
+    # forcefully open youtube in mobile mode
+    driver.get('https://m.youtube.com/?persist_app=1&app=m')
+    # list of urls for mobile emulation
+    list_of_urls: list = [
+        "https://m.youtube.com/watch?v=IpeXtG9sBEI"
+    ]
+
+    for index, url in enumerate(list_of_urls):
+        # this stores the complete information of the video being played
+        video_info_details: dict = {}
+        # logs any errors (if found)
+        error_list: list = []
+        # keeps track of the count of unique ads displayed during the main video
+        unique_add_count: int = 0
+        # boolean set to False -- tracks whether an ad is being displayed or not
+        ad_just_played: bool = False
+        # stores the buffer details at every second
+        buffer_list: list = []
+        actual_buffer_reads: list = []
+        buffer_size_with_ad: list = []
+        # stores the resolution of the main video at every second -- if the video is
+        # a 100 seconds long then the length of this list would be 100
+        # this can stretch to as large at 3599 (>= 3600 is not allowed)
+        vid_res_at_each_second: list = []
+        main_res_all: list = []
+        # this stores the ad id of the recent most ad played
+        previous_ad_id: str = url.split("=")[1]
+
+        movie_id: str = url.split("=")[1]
+        # stores the name of the directory to store the videos in
+        new_dir: str = "./" + str(index + 1)
+
+        # spawns a chrome instance and directs to the url in the list as per the iteration of the loop
+        # now direct to the url 
+        driver.get(url)
+
+        # ==================================================================================================
+        # navigating using XMLpaths
+        # Playing Video
+        movie_player = driver.find_element_by_id("movie_player")
+        hover = ActionChains(driver).move_to_element(movie_player)
+        hover.perform()
+        ActionChains(driver).context_click(movie_player).perform()
+
+        # Clicking Hamburger Icon
+        settings = driver.find_element_by_xpath(
+            "/html/body/ytm-app/ytm-mobile-topbar-renderer/header/div/ytm-menu/button")
```

Comment:
Not a fan of Xpath, they change and in my experience have caused crashes. Classnames don't change.